File: CSI2120/assignment/comprehensive/part2/src/student/student.go
package student

import "person"

// Student is a struct to represent students
type Student struct {
	P *person.Person
}

// NewStudent is the constructor of Student struct
func NewStudent(p *person.Person) *Student {
	return &Student{p}
}

func (s *Student) Prefer(name string) bool {
	var index int64
	for i := 0; i < len(s.P.Pref); i++ {
		if s.P.Pref[i] == name {
			index = int64(i)
		}
	}
	return index < s.P.Match
}

File: Archive/CSI2120/assignment/comprehensive/part1/Student.java
/**
 * Student 
 * Hongyi Lin, 300053082
 */
public class Student extends Person {
    // constructor
    public Student(String name, int n) {
        super(name, n);
    }
    public Student(String name, String[] pref) {
        super(name, pref);
    }
    // method
    /*
     * return true if this student prefer the new employer with the name other than the current
     */
    public boolean prefer(String other) {
        int index = -1;
        for (int i = 0; i < pref.length; ++i) {
            if (pref[i].equals(other)) {
                index = i;
            }
        }
        return index < match;
    }
}